### DISCUSSION TITLE
Help configuring project with new MS code coverage option

### DISCUSSION THREAD

**Discussion by fossbrandon** [FortuneN/FineCodeCoverage#261] [Q&A] [answered]
I've had my repo setup to use the "old method" of configuring FCC but I want to try and get it setup to use MS code coverage since it meets my use cases better. However, I've had mixed success and I'm hoping to get some help.

I've been able to remove my existing `runsettings` file, configure my test projects csproj files to specify `<RunMsCodeCoverage>Yes</RunMsCodeCoverage>` within `<PropertyGroup Label="FineCodeCoverage">`, and then run one test class and have it properly use MS code coverage to only produce coverage results for the code affected by the one test class. This is awesome!

However, I want to be able to configure my own `runsettings` file but I'm having trouble following the readme. I am new to these concepts though so I could just be naïve. I've specified `<RunMsCodeCoverage>IfInRunSettings </RunMsCodeCoverage>` within `<PropertyGroup Label="FineCodeCoverage">` inside each of my test project csproj files, installed the `Microsoft.CodeCoverage` nuget package within each test project, and then created a `runsettings` file in the same directory as my root `.sln`. However, I'm not sure what path to specify for `<TestAdaptersPaths>`. When I had FCC generate my runsettings file, it looks like it points to an FCC MS code coverage folder. However, if I install the nuget packages myself, should I be pointing somewhere within my own project? Maybe my bin folder? Regardless of what I put in there, it still seems to generate its own file rather than use my `runsettings` file. Any advice would be greatly appreciated, thanks!

FYI: I'm not sure if this is a problem, but even when MS code coverage appears to work, I get the following error message in the FCC output.
```
Fine Code Coverage : Failed to override 'RunMsCodeCoverage' setting
System.Exception: Cannot handle 'RunMsCodeCoverage' yet
   at FineCodeCoverage.Engine.Model.SettingsMerger.GetValueFromXml(XElement settingsElement, PropertyInfo property)
   at FineCodeCoverage.Engine.Model.SettingsMerger.TryGetValueFromXml(XElement settingsElement, PropertyInfo property)

Fine Code Coverage : Ms code coverage

Fine Code Coverage : ReportGenerator Run Arguments [reporttype:Cobertura] 
.
.
.
```

**Comment by tonyhallett** [comment, marked as answer]
Firstly the exception you encountered.  The RunMsCodeCoverage option has to apply to all the test projects and as such should only be specified as a vs option.  I will update the docs.

With regards to the test adapter - this needs to point to the location of the dll.  For the one that comes with FCC this is 
C:\Users\YOU\AppData\Local\FineCodeCoverage\msCodeCoverage\17.1.0\build\netstandard1.0.
If you are installing the nuget package then you should use your nuget installation path - C:\Users\YOU\.nuget\packages\microsoft.codecoverage

You did not need to install the package as FCC will provide the TestAdaptersPaths to the test framework ( without touching your runsettings file )

**Reply by fossbrandon** [reply]
Awesome, good to know. It would be a nice enhancement down the road to allow the ms code coverage option to be specified somewhere in the repo so that anyone with the extension using the repo will collect code coverage the same way. Thank you!
